dvc/utils/serialize/_yaml.py

```diff
@@ -14,14 +15,77 @@ def __init__(self, path):
         super().__init__(path, "YAML file structure is corrupted")
 
 
-def load_yaml(path, tree=None):
-    return _load_data(path, parser=parse_yaml, tree=tree)
+def recursive_render(tpl, values, max_passes=100):
+    """This is a bit of black magic to recursivly render a
+    template. Adaped from:
+
+      https://stackoverflow.com/questions/8862731/jinja-nested-rendering-on-variable-content
+
+    Args:
+      tpl: Template string
+      values: dict of values. Importantly this dict can contain
+          values that are themselves {{ placeholders }}
+      max_passes: Limits the number of times we loop over the
+          template.
+
+    Returns:
+      rendered template.
+    """
+    prev = tpl
+    for _ in range(max_passes):
+        curr = Template(prev).render(**values)
+        if curr != prev:
+            prev = curr
+        else:
+            return curr
+    raise RecursionError("Max resursion depth reached")
+
+
+def render_vars(dvc_dict):
```

Comment:
@tall-josh, this interpolates values in `vars` using data from `vars` itself, right?
```jinja
vars:
  foo: 3
  bar: {{ foo }}
```

Reply:
How important is this scenario?

Reply:
Sorry for the delay. Yeah it does. I have used this a lot. ie:

```yaml
vars:
  my_labeling_project: 123
  data_out: /storage/data/{{ my_labeling_project }}_data
  src: s3://labeled-data/projects/{{ my_labeling_project }}
stages:
  download:
    cmd: python download_data.py --src {{src}} --out  {{ data_out }}
    outs:
      - {{ data_out }}
```

Without this I would have to repeat the `my_labeling_project`